### PR TITLE
Handle pending transaction

### PR DESCRIPTION
**Pull request by kolya182**
Fixes https://github.com/trustwallet/blockatlas/issues/683

Now pending transactions returned

### CODE REVIEW

File: platform/bitcoin/model.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ type Transaction struct {
 	Vin           []Output `json:"vin"`
 	Vout          []Output `json:"vout"`
 	BlockHash     string   `json:"blockHash"`
-	BlockHeight   uint64   `json:"blockHeight"`
+	BlockHeight   int64    `json:"blockHeight"`
```
Comment (hewigovens) [inline platform/bitcoin/model.go]: why int64? negative block number?

File: platform/bitcoin/api.go [hunk 2]
```diff
@@ -305,3 +300,11 @@ func (p *Platform) InferValue(tx *blockatlas.Tx, direction blockatlas.Direction,
 	}
 	return value
 }
+
+func (trx *Transaction) getTransactionStatus() blockatlas.Status {
+	if trx.Confirmations == 0 {
+		return blockatlas.StatusPending
+	} else {
+		return blockatlas.StatusCompleted
```
Comment (vikmeup) [inline platform/bitcoin/api.go]: why `else`?

just `return blockatlas.StatusCompleted`

File: platform/bitcoin/api_test.go [hunk 3]
```diff
@@ -262,3 +263,27 @@ func TestInferDirection(t *testing.T) {
 		}
 	}
 }
+
+func TestTransactionStatus(t *testing.T)  {
+	var trxPending = `{"confirmations": 0}`
+	var trxCompleted = `{"confirmations":1}`
+
```
Comment (vikmeup) [inline platform/bitcoin/api_test.go]: why? 

just use `Transaction{Confirmations: 1}`

File: platform/bitcoin/api.go [hunk 4]
```diff
@@ -305,3 +300,11 @@ func (p *Platform) InferValue(tx *blockatlas.Tx, direction blockatlas.Direction,
 	}
 	return value
 }
+
+func (trx *Transaction) getTransactionStatus() blockatlas.Status {
```
Comment (vikmeup) [inline platform/bitcoin/api.go]: `getTransactionStatus ` => `Status` or `GetStatus`

File: platform/bitcoin/model.go (hunk 1 above)
Comment (kolya182) [reply platform/bitcoin/model.go]: Yes, when transaction is pending block hight = `-1`

File: platform/bitcoin/api.go (hunk 2 above)
Comment (hewigovens) [reply platform/bitcoin/api.go]: in this case, we should check block numbers negative here?

Comment (kolya182) [reply platform/bitcoin/api.go]: I has this check https://github.com/trustwallet/blockatlas/pull/684/commits/3b69f7f9d44c24a25b743c544d0681a751aa6054#diff-7bc94df832258a11b2a6d0a76bf76c87R200 but removed, one condition should be sufficient

File: platform/bitcoin/api.go [hunk 5]
```diff
@@ -305,3 +300,10 @@ func (p *Platform) InferValue(tx *blockatlas.Tx, direction blockatlas.Direction,
 	}
 	return value
 }
+
+func (trx *Transaction) getStatus() blockatlas.Status {
```
Comment (vikmeup) [inline platform/bitcoin/api.go]: `trx` = `tx`

File: platform/bitcoin/api_test.go [hunk 6]
```diff
@@ -262,3 +263,17 @@ func TestInferDirection(t *testing.T) {
 		}
 	}
 }
+
+func TestTransactionStatus(t *testing.T) {
+	var tests = []struct {
+		Trx      Transaction
```
Comment (vikmeup) [inline platform/bitcoin/api_test.go]: `Trx` = `Tx`

**Comment by kolya182** [comment]
Test fails due to slow response from aion explorer, reported to the team to fix it, merging for now

**Pull request merged**